### PR TITLE
Document filter usage in image building

### PR DESCRIPTION
**Pull request by pwnall**
This documents the code changes in #1310

### CODE REVIEW

**Comment by pwnall** [comment]
I look forward to feedback!

I reused the Dockerfile content in the [main Docker tutorial](https://docs.docker.com/userguide/dockerimages/#building-an-image-from-a-dockerfile), but replaced the `touch` + text block with a series of `echo` commands. I'd be particularly interested in learning if there's a better way to get a Dockerfile set up, for doc purposes.

**Comment by abronan** [comment]
Thanks @pwnall! This is super useful!

LGTM

**Comment by dongluochen** [comment]
LGTM. thanks @pwnall!

**Comment by aluzzardi** [comment]
LGTM

**Comment by aluzzardi** [comment]
ping @moxiegirl

File: docs/scheduler/filter.md [hunk 1]
```diff
@@ -90,6 +91,38 @@ Now we want to run an Nginx frontend in our cluster. However, we don't want
 
 The scheduler selected `node-2` since it was started with the `storage=disk` label.
 
+Last, let's build a custom Sinatra image. We'll avoid flash drives once again.
```
Comment (moxiegirl) [inline docs/scheduler/filter.md]: Finally, build a custom Sinatra image. Again, you'll avoid flash drives.

File: docs/scheduler/filter.md [hunk 2]
```diff
@@ -28,9 +28,10 @@ You can choose the filter(s) you want to use with the `--filter` flag of `swarm
 Constraints are key/value pairs associated to particular nodes. You can see them
 as *node tags*.
 
-When creating a container, the user can select a subset of nodes that should be
-considered for scheduling by specifying one or more sets of matching key/value
-pairs. This approach has several practical use cases such as:
+When creating a container or building an image, the user can select a subset of
```
Comment (moxiegirl) [inline docs/scheduler/filter.md]: @pwnall Thank you for the contribution.  The structure can be tightened by doing this:

```markdown
When creating a container or building an image, you can identify a subset of
nodes to consider for scheduling. This approach has several practical use cases such as:

* Selecting specific host properties (such as `storage=ssd`, in order to schedule
  containers on specific hardware).
* Tagging nodes based on their physical location (`region=us-east`, to force
  containers to run on a given location).
* Logical cluster partitioning (`environment=production`, to split a cluster into
  sub-clusters with different properties).

To specify a subset of key/value pairs for a node, pass a list of `--label`
options at docker startup time. For instance, let's start `node-1` with the
`storage=ssd` label:
```

Which yields this: 

----

When creating a container or building an image, you can identify a subset of
nodes to consider for scheduling. This approach has several practical use cases such as:

* Selecting specific host properties (such as `storage=ssd`, in order to schedule
  containers on specific hardware).
* Tagging nodes based on their physical location (`region=us-east`, to force
  containers to run on a given location).
* Logical cluster partitioning (`environment=production`, to split a cluster into
  sub-clusters with different properties).

To specify a subset of key/value pairs for a node, pass a list of `--label`
options at docker startup time. For instance, let's start `node-1` with the
`storage=ssd` label:

**Comment by pwnall** [comment]
@moxiegirl Thank you very much for the much improved writing!

I applied your suggestions, can you please take another look?

**Comment by moxiegirl** [comment]
@pwnall You are quick! Thank you LGTM

**Comment by pwnall** [comment]
@moxiegirl Thank *you* for the great feedback! :heart: docker

**Comment by moxiegirl** [comment]
@pwnall Thank you very much for the contribution! Docker :heart:s you for contributing and for being good at it too!

**Pull request merged**